src/lib/datepicker/calendar.ts

```diff
@@ -29,7 +30,119 @@ import {MatDatepickerIntl} from './datepicker-intl';
 import {MatMonthView} from './month-view';
 import {MatMultiYearView, yearsPerPage} from './multi-year-view';
 import {MatYearView} from './year-view';
+import {ComponentPortal, ComponentType, Portal} from '@angular/cdk/portal';
+import {Subject} from 'rxjs/Subject';
+import {takeUntil} from 'rxjs/operators/takeUntil';
 
+/** Default header for MatCalendar */
+@Component({
+  moduleId: module.id,
+  selector: 'mat-calendar-header',
+  templateUrl: 'calendar-header.html',
+  encapsulation: ViewEncapsulation.None,
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MatCalendarHeader implements OnDestroy {
+  /** Subject that emits when the component has been destroyed. */
+  private _destroyed = new Subject<void>();
+
+  constructor(private _intl: MatDatepickerIntl,
+              @Host() public calendar: MatCalendar<any>,
+              @Optional() private _dateAdapter: DateAdapter<any>,
+              @Optional() @Inject(MAT_DATE_FORMATS) private _dateFormats: MatDateFormats,
+              changeDetectorRef: ChangeDetectorRef) {
+    _intl.changes.pipe(takeUntil(this._destroyed)).subscribe(
```

Comment:
nit: format like this:

```ts
_intl.changes.pipe(takeUntil(this._destroyed))
    .subscribe(() => changeDetectorRef.markForCheck());
```

Reply:
done in 6d0a4d2264e085c011770bf3de0fe381598dbe33